src/System.Text.Json/tests/JsonDocumentTests.cs

```diff
@@ -3740,6 +3671,451 @@ private static string GetCompactJson(TestCaseType testCaseType, string jsonStrin
 
             return s_compactJson[testCaseType] = existing;
         }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public static void WriteNumber(bool indented)
+        {
+            WriteSimpleValue(indented, "42");
```

Comment:
Rather than duplicate all of these tests it'd be better to make them reusable.  Given that we're on sort of a short timeline here I'd suggest just doing two tests for JsonDocument:

1) `AssertExtensions.Throws<ArgumentNullException>("writer", doc.WriteTo(null));`
2) One simple test that reads `"[ 1e400 ]"` and writes `"[1e400]"` using the default (non-indented) mode of the writer.


If you want to tackle reusability, that'd be
* Change `public static class JsonElementWriteTests` to `public abstract class JsonDomWriteTests`
* Add something like `public abstract void Write(JsonDocument document, Utf8JsonWriter writer);`
* Make WriteSimpleValue use the abstract method instead of `target.WriteValue(writer);`
* Maybe also add a `public abstract bool ShouldWrap { get; }`?
* Use ShouldWrap appropriately.
* Make two new types at the top of the file

```C#
public class JsonElementWriteTests : JsonDomWriteTests
{
    public override bool ShouldWrap => true;
    public override void Write(JsonDocument document, Utf8JsonWriter writer) => document.RootElement[0].WriteTo(writer);
}

public class JsonDocumentWriteTests : JsonDomWriteTests
{
    public override bool ShouldWrap => false;
    public override void Write(JsonDocument document, Utf8JsonWriter writer) => document.WriteTo(writer);
}
```

* Make sure the tests are actually still executing by making one fail (you might have to remove the word static from all of them once you're using inheritance... I forget)
* Fix the failure before pushing the new commits :)

Reply:
(and any tests which were specific to using WriteTo from `default(JsonElement)` would then get moved into JsonElementWriteTests)

Reply:
I had other confusion too. Since `document.WriteTo` inturn calls `RootElement.WriteTo`, do we need to test all those scenario again? 

Going for reusability now might get complicated I think. I will work on this as separate PR if needed. For now, I will remove all those tests and add simple test as you mentioned. Is that fine?

Reply:
> For now, I will remove all those tests and add simple test as you mentioned. Is that fine?

Yep.

Reply:
Done. Removed those tests and added one simple test for now.